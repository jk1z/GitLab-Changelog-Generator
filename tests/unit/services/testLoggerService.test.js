const MockDate = require("mockdate");
const LoggerService = require("../../../app/services/logger");
describe("Logger service", () => {
    let loggerService;
    beforeEach(() => {
        MockDate.set("2018-09-07T11:16:17.520Z");
        jest.resetAllMocks();
        loggerService = new LoggerService({ config: { NODE_ENV: "local", TZ: "Australia/Melbourne" } });
        loggerService.console.log = jest.fn();
    });

    afterEach(() => {
        MockDate.reset();
    });

    test("should log info", () => {
        loggerService.info(`test`);
        expect(loggerService.console.log).toBeCalledTimes(1);
        expect(loggerService.console.log.mock.calls[0][0]).toMatchSnapshot();
    });
    test("should log warn", () => {
        loggerService.warn(`test`);
        expect(loggerService.console.log).toBeCalledTimes(1);
        expect(loggerService.console.log.mock.calls[0][0]).toMatchSnapshot();
    });
    test("should log debug", () => {
        loggerService.debug(`test`);
        expect(loggerService.console.log).toBeCalledTimes(1);
        expect(loggerService.console.log.mock.calls[0][0]).toMatchSnapshot();
    });
    test("should log error", () => {
        loggerService.error(`test`);
        expect(loggerService.console.log).toBeCalledTimes(1);
        expect(loggerService.console.log.mock.calls[0][0]).toMatchSnapshot();
    });
});
