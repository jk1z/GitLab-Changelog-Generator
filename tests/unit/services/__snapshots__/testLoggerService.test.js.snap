// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger service should log debug 1`] = `"[97m2018-09-07T21:16:17+10:00 [DEBUG] test[39m"`;

exports[`Logger service should log error 1`] = `"[91m2018-09-07T21:16:17+10:00 [ERROR] test[39m"`;

exports[`Logger service should log info 1`] = `"[92m2018-09-07T21:16:17+10:00 [INFO] test[39m"`;

exports[`Logger service should log warn 1`] = `"[93m2018-09-07T21:16:17+10:00 [WARN] test[39m"`;
